### PR TITLE
command/server: add VAULT_CLUSTER_NAME environment variable support

### PR DESCRIPTION
**Pull request by jasonodonnell**
This adds a new environment variable `VAULT_CLUSTER_NAME` support for the [cluster_name](https://developer.hashicorp.com/vault/docs/configuration#cluster_name) configurable. This makes it easier to configure the `cluster_name` in our Vault Helm project. 

If the leader changes without `cluster_name` configured, the new leader will call [NewClusterMetricSink](https://github.com/hashicorp/vault/blob/main/helper/metricsutil/wrapped_metrics.go#L124-L132) and cluster name will be empty for some time. If you're parsing telemetry this will lead to noise in the data:

```
vault_core_replication_dr_primary{cluster="",host="vault-1"} 0
vault_core_replication_dr_primary{cluster="vault-cluster-c24ee36f",host="vault-1"} 0
```

### CODE REVIEW

File: command/server.go [hunk 1]
```diff
@@ -1106,6 +1106,15 @@ func (c *ServerCommand) Run(args []string) int {
 
 	logProxyEnvironmentVariables(c.logger)
 
+   	if clusterName := os.Getenv("VAULT_CLUSTER_NAME"); clusterName != "" {
+		var err error
+		config.ClusterName = clusterName
+		if err != nil {
```
Comment (raskchanky) [inline command/server.go:1112]: I'm guessing this whole if statement can be omitted, since err is never assigned to anything, so it will always be nil.

**Review by raskchanky — APPROVED** [review]

File: command/server.go (hunk 1 above)
Comment (jasonodonnell) [reply command/server.go:1112]: Whoops that's what I get for copying an env with with Parsebool 😅

Comment (raskchanky) [reply command/server.go:1112]: Your commit to remove the error check didn't actually remove it. 😅

**Comment by maxb** [comment]
> For helm it's much easier to set the `cluster_name` to be the name of the pod

Wouldn't it be really unexpected for the cluster_name to be set to the name of the pod, rather than the name of the cluster? Also, wouldn't that mess up the metrics by changing the value depending on which pod takes leadership?

**Comment by jasonodonnell** [comment]
> > For helm it's much easier to set the `cluster_name` to be the name of the pod
> 
> Wouldn't it be really unexpected for the cluster_name to be set to the name of the pod, rather than the name of the cluster? Also, wouldn't that mess up the metrics by changing the value depending on which pod takes leadership?

@maxb On closer inspection that's correct and I think I found a different bug.

**Pull request closed without merging**